src/container/srv_target.c

```diff
@@ -1452,14 +1453,16 @@ cont_child_aggregate(struct ds_cont_child *cont, uint64_t *sleep)
 	 * sc_aggregation_min != DAOS_EPOCH_MAX means we need to aggregate
 	 * from epoch 0 because some snapshot was deleted.
 	 */
-	if (cont->sc_aggregation_min != DAOS_EPOCH_MAX) {
+	if (cont->sc_aggregation_min != DAOS_EPOCH_MAX ||
+	    cont->sc_pool->spc_reset_aggregation) {
 		epoch_min = 0;
 		/*
 		 * Set sc_aggregation_min to non-zero so that we can tell
 		 * if another snapshot deletion happened when this round of
 		 * aggregation done.
 		 */
 		cont->sc_aggregation_min = hlc;
+		cont->sc_pool->spc_reset_aggregation = 0;
```

Comment:
What if cont_child_aggregate() returns before reaching the function ending? I'm wondering if we should just keep the aggregate epoch min unchanged in this case, and add only one more range of [rebuild prev_snap + 1, rebuild next_snap]?